gradio/components/chatbot.py

```diff
@@ -287,54 +398,107 @@ def create_file_message(chat_message, filepath):
+                f"Invalid message for Chatbot component: {message}", visible=False
+            )
+
+        # extract file path from message
+        new_messages = []
+        if isinstance(msg.content, str):
+            for word in msg.content.split(" "):
+                filepath = Path(word)
+                try:
+                    is_file = filepath.is_file() and filepath.exists()
+                except OSError:
+                    is_file = False
+                if is_file:
+                    filepath = cast(
+                        str, move_resource_to_block_cache(filepath, block=self)
+                    )
+                    mime_type = client_utils.get_mimetype(filepath)
+                    new_messages.append(
+                        Message(
+                            role=msg.role,
+                            metadata=msg.metadata,
+                            content=FileMessage(
+                                file=FileData(path=filepath, mime_type=mime_type)
+                            ),
+                        ),
+                    )
+        return [msg, *new_messages]
+
     def postprocess(
         self,
-        value: (
-            list[
-                list[str | GradioComponent | tuple[str] | tuple[str, str] | None]
-                | tuple
-            ]
-            | None
-        ),
-    ) -> ChatbotData:
+        value: TupleFormat | list[MessageDict | Message] | None,
+    ) -> ChatbotDataTuples | ChatbotDataMessages:
         """
         Parameters:
             value: expects a `list[list[str | None | tuple]]`, i.e. a list of lists. The inner list should have 2 elements: the user message and the response message. The individual messages can be (1) strings in valid Markdown, (2) tuples if sending files: (a filepath or URL to a file, [optional string alt text]) -- if the file is image/video/audio, it is displayed in the Chatbot, or (3) None, in which case the message is not displayed.
         Returns:
             an object of type ChatbotData
         """
         if value is None:
-            return ChatbotData(root=[])
-
-        processed_messages = []
-        for message_pair in value:
-            if not isinstance(message_pair, (tuple, list)):
-                raise TypeError(
-                    f"Expected a list of lists or list of tuples. Received: {message_pair}"
-                )
-            if len(message_pair) != 2:
-                raise TypeError(
-                    f"Expected a list of lists of length 2 or list of tuples of length 2. Received: {message_pair}"
-                )
-            processed_messages.append(
-                [
-                    self._postprocess_chat_messages(message_pair[0]),
-                    self._postprocess_chat_messages(message_pair[1]),
-                ]
-            )
-        return ChatbotData(root=processed_messages)
+            return ChatbotDataTuples(root=[])
+        if self.msg_format == "tuples":
```

Comment:
Note: for all of our other components that can accept different types (e.g. `Image` with multiple types such as PIL images, numpy arrays, string filepaths), we don't enforce the `type` in `postprocess()` -- we just automatically figure it out based on `value`. Potentially, we could do the same thing here

Reply:
We use the msg_format to set the data model and api info. So I want to make sure it matches in postprocess. Otherwise, it would be possible for the API info to tell you to expect dictionaries but you get tuples.